conda_build/main_develop.py

```diff
@@ -32,6 +36,12 @@ def main():
         nargs='+',
         help="Path to the source directory."
     )
+    p.add_argument(
+                   '-l', '--post-link',
```

Comment:
Currently, for conda develop it should just work as you suggested.
If --post-link is explicitly given, then it only does the relinking, so:

if not args.post_link:
    write_to_conda_pth()

# next step is to do the relinking for shared objects
relink_sharedobjects()

I just tested it on my example, by doing the following in CythonExample/:
./setup.py build_ext --inplace
conda develop .

The conda develop appends to conda.pth, then it does the relinking.